### DISCUSSION TITLE
Use of new row sorting _dnd=

### DISCUSSION THREAD

**Discussion by djbbiz** [cortexrd/Knack-Toolkit-Library#184] [General]
@cortexrd For the new _dnd= feature can you use this if you have Fields that are Grouped to provide headings in the Grid? Also can you still sort on secondary or tertiary fields if the Sort Order field is the first priority and the other sort columns are listed after?

**Comment by cortexrd** [comment]
Hi Dean,

Currently, none of this is supported properly.

I should have mentioned it clearly in the documentation, and added defensive coding to disable the _dnd keyword when groups are found.

But I could find a way to make it work.  How would you see it?

Here's my idea:

- If groups are found _dnd would work within that group. The numbering would be from 1 to n inside each group.
- If we have more than one group, same thing.
- This way, if you decide to change the sort order of any group, all should remain in place since the groups would be re-ordered but not the rows inside each.

Let me know if I forgot something,
Norm

**Comment by cortexrd** [comment]
@djbbiz 

I improved the **_dnd** keyword.
You can now perform dnd within **groups**.

The syntax is the same, groups are auto-detected.

When you re-order rows in a group, they will be renumbered from 1 to n within that group as it should (instead of through the whole list).

I tested with one grouping level only.  If your setup has more, let me know if it works.

Enjoy,
Norm

**Comment by cortexrd** [comment]
You can test it in my new demo pages:

https://ctrnd.knack.com/ktl-tutorials#dragn-drop/

No login required.

Norm

**Comment by LeighBerrell** [comment]
Norm,
Are you considering adding the ability to suppress _dnd when on a mobile device?  I find that on small mobile screens its quite easy to accidentally reorder rows when you really are trying to scroll across the row.

**Comment by Deanobeano2** [comment]
Norm

I second Leigh's suggestion as it was frighteningly easy to re-order the rows on a smart phone. Very impressive performance visually.

Dean

**Comment by cortexrd** [comment]
Hi guy,

I will check into this.  There are at least 3 ways to address the issue:

1. Add a delay before the drag starts (good and easy)
2. Add a "handle" to each row where you must click to drag, elsewhere will be just a normal scroll (more work)
3. Add minimum distance before drag starts (not good)

Disabling on mobile is not desirable, I think.  I'll try anything else before doing this.

Norm

**Comment by cortexrd** [comment]
Hi guys,

I fixed the issue by adding a delay on mobile devices before the drag starts.  The library had support just for that scenario - which was awesome!

So now, when you touch the screen, you must wait 500ms before dragging the row.  This allows smooth scrolling without problem.

Test the **dev** version please and let me know if it works with your device.
Tested on a Galaxy Tab A and Samsung S23 Ultra.

Norm
